### DISCUSSION TITLE
What is the recommended equivalent to `git grep`

### DISCUSSION THREAD

**Discussion by phlax** [gitpython-developers/GitPython#1404] [Q&A] [answered]
Im shifting some code from bash to python that makes use of `git grep`

Im wondering if there is a recommended way to do this - eg with a `predicate` in traverse or using `execute`

Im particularly keen to replicate git grep's `-I` functionality which filters out "binary" files from gits pov (ie .gitattributes)

**Comment by Byron** [comment, marked as answer]
`GitPython` won't  help much here as it merely acts as wrapper around `git` to allow something like `repo.git.grep(I=true)`, the result has to be parsed by hand. There is no other support for `grep` like functionality or gitattributes.
